src/ck-core/ckmarshall.h

```diff
@@ -1,18 +1,47 @@
 #ifndef _CKMARSHALL_H_
 #define _CKMARSHALL_H_
 
-#include "charm++.h"
+// This is the message type marshalled parameters get packed into:
+class CkMarshallMsg;
+class CkEntryOptions;
+
+CkMarshallMsg *CkAllocateMarshallMsg(int size, const CkEntryOptions *opts = nullptr);
+
+namespace ck {
+using marshall_msg = CkMarshallMsg *;
+
+inline char *get_message_buffer(marshall_msg msg);
+
+template <class... Args>
+inline marshall_msg make_marshall_message(Args... args) {
+  using expander = int[];
+  PUP::sizer s;
+  (void)expander{0, (void(s | args), 0)...};
```

Comment:
Two things:

1. This expansion stuff needs an explanatory comment, it's too hard to understand otherwise.
2. This should probably be pushed into PUP, it should have a way of dealing with a parameter pack.

Reply:
1) I originally had a few links here, will restore them.
2) Would something like `PUP::many(PUP::er &p, Args... args)` be OK?

Reply:
1. :+1: 
2. That, or just implementing `operator|` and `operator()` on a parameter pack.